### PR TITLE
(SR-779) Add search-by-ID and release display to Linear app

### PR DESCRIPTION
**Pull request by dsayerdp**
## Summary

Two features split out of the SR-91 working branch because they go beyond that bug's scope (the SR-91 relationship fix itself is in #84):

### 1. Search issues by ID
- Replaces the `title containsIgnoreCase` filter with Linear's `searchIssues(term:)` GraphQL query, so agents can find issues by identifier (e.g. `ABC-123`) as well as title text.
- `gql()` now omits the `variables` key from the POST body when empty — Linear rejects an empty `variables` object.

### 2. Display releases
- New `ReleaseItem` component and a `Releases` block on the issue view.
- `Releases` property on issue list items (home page).
- `releases { id name version url stage { ... } }` added to the issue GraphQL selections, with hand-rolled `Release`/`ReleaseStage` types (the vendored schema predates Linear's Releases feature).

## Notes
- Deliberately excludes the SR-91 fix so the two can land independently. Expect small merge conflicts with #84 in `getIssuesService.ts`, `getIssueService.ts`, `types.ts`, and `manifest.json` (both bump to 1.0.25) — whichever merges second resolves them.

## Testing
- `tsc --noEmit` clean, `pnpm lint` clean, `pnpm test`: 41 suites / 151 tests pass.
- New unit tests: `getIssuesService` (searchIssues term + id-filter + releases selection), `getIssueService` (releases selection), `gql` (empty variables omitted), `IssueItem` (renders linked releases).

Linear: https://linear.app/deskpro/issue/SR-779/linear-app-search-issues-by-id-and-display-releases

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Add issue search by identifier and surface Linear releases on issue views and listings while improving GraphQL request handling.

New Features:
- Support searching Linear issues by free-text term that matches both identifiers and titles.
- Display associated releases on individual issue views and issue list items, including release link and stage details.

Enhancements:
- Extend Linear issue types with Release and ReleaseStage to support the Releases feature not present in the vendored schema.
- Refine GraphQL query construction to reuse shared issue field selections and omit empty variables from requests.
- Introduce pnpm workspace configuration and move dependency override configuration from package.json to workspace-level settings.

Build:
- Bump Linear app manifest version to 1.0.25.

Tests:
- Add unit tests for getIssuesService and getIssueService covering search-by-term, id filtering, and releases selection in GraphQL queries.
- Add unit test for gql helper to verify empty variables objects are omitted from payloads.
- Extend IssueItem tests to verify rendering of linked releases in the UI.

### CODE REVIEW

File: src/services/linear/getIssuesService.ts [hunk 1]
```diff
@@ -17,48 +17,92 @@ type Params = {
...
+        id
+        name
+        type
+        color
+      }
+    }
+  }
+`;
 
+const getIssuesService = (client: IDeskproClient, params?: Params) => {
```
Comment (AshleyDawson) [inline src/services/linear/getIssuesService.ts:80]: Optional (see bottom of this comment).

Recommend we double-check the return shape of `searchIssues` here?

This query is selecting `id`, `identifier`, `title`, `priority`, etc. directly from `searchIssues(term: $term).nodes`, but the tests also note that this returns `IssueSearchResult` rather than `Issue`. If that result can contain mixed node types or needs narrowing via an inline fragment, this may break at runtime or return partial data.

If the schema supports multiple result types, I think we should guard this with something like:

```graphql
searchIssues(term: $term) {
  nodes {
    ... on Issue {
      id
      identifier
      title
      priority
      priorityLabel
      url
      dueDate
      ...
    }
  }
}
```

If the live schema guarantees `Issue` here, a short comment explaining that assumption would also help future readers.

**Review by AshleyDawson — APPROVED** [review]
Approved with an optional change request
